py3dtiles/gltf.py

```diff
@@ -0,0 +1,417 @@
+    if(math.fabs(vectProd[0]) > math.fabs(vectProd[1]) and math.fabs(vectProd[0]) > math.fabs(vectProd[2])):
+        # (yz) projection
+        for v in range(0,len(polygon)):
+            polygon2D.append([polygon[v][1], polygon[v][2]])
+    elif(math.fabs(vectProd[1]) > math.fabs(vectProd[2])):
+        # (zx) projection
+        for v in range(0,len(polygon)):
+            polygon2D.append([polygon[v][0], polygon[v][2]])
+    else:
+        # (xy) projextion
+        for v in range(0,len(polygon)):
+            polygon2D.append([polygon[v][0], polygon[v][1]])
+
+    triangulation = triangle.triangulate({'vertices': polygon2D, 'segments': segments})
+    if 'triangles' not in triangulation:    # if polygon is degenerate
+        return []
+    trianglesIdx = triangulation['triangles']
+    triangles = []
+
+    for t in trianglesIdx:
+        # triangulation may break triangle orientation, test it before adding triangles
+        if(t[0] > t[1] > t[2] or t[2] > t[0] > t[1] or t[1] > t[2] > t[0]):
+            triangles.append([polygon[t[1]], polygon[t[0]],polygon[t[2]]])
+        else:
+            triangles.append([polygon[t[0]], polygon[t[1]],polygon[t[2]]])
+
+    return triangles
+
+
+def compute_normals(triangles):
+    normals = []
+    for t in triangles:
+        U = t[1] - t[0]
+        V = t[2] - t[0]
+        N = np.cross(U,V)
+        norm = np.linalg.norm(N)
+        if norm == 0:
+            normals.append(np.array([1,0,0], dtype=np.float32))
+        else:
+            normals.append(N / norm)
+    return normals
+
+
+def parse(wkb):
+    """
+    Expects Multipolygon Z
+    """
+    multiPolygon = []
+    #length = len(wkb)
+    #print(length)
+    #byteorder = struct.unpack('b', wkb[0:1])
+    #print(byteorder)
+    #geomtype = struct.unpack('I', wkb[1:5])    # 1006 (Multipolygon Z)
+    #print(geomtype)
+    geomNb = struct.unpack('I', wkb[5:9])[0]
+    #print(geomNb)
+    #print(struct.unpack('b', wkb[9:10])[0])
+    #print(struct.unpack('I', wkb[10:14])[0])   # 1003 (Polygon)
+    #print(struct.unpack('I', wkb[14:18])[0])   # num lines
+    #print(struct.unpack('I', wkb[18:22])[0])   # num points
+    offset = 9
+    for i in range(0, geomNb):
+        offset += 5#struct.unpack('bI', wkb[offset:offset+5])[0]  # 1 (byteorder), 1003 (Polygon)
+        lineNb = struct.unpack('I', wkb[offset:offset+4])[0]
+        offset += 4
+        polygon = []
+        for j in range(0, lineNb):
+            pointNb = struct.unpack('I', wkb[offset:offset+4])[0]  # num points
+            offset += 4
+            line = []
+            for k in range(0, pointNb-1):
+                point = np.array(struct.unpack('ddd', wkb[offset:offset+24]), dtype=np.float32)
+                offset += 24
+                line.append(point)
+            offset += 24   # skip redundant point
+            polygon.append(line)
+        multiPolygon.append(polygon);
+    return multiPolygon;
+
+# TODO: remove
+import binascii
+if __name__ == "__main__":
```

Comment:
The `main` should be removed from this file. Maybe you could add this part to unit tests?